src/mod/pbkdf2.mod/pbkdf2.c

```diff
@@ -0,0 +1,302 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * pbkdf2.c -- part of pbkdf2.mod
+ *
+ * Written by thommey and Michael Ortmann
+ *
+ * Copyright (C) 2017 - 2020 Eggheads Development Team
+ */
+
+#include "src/mod/module.h"
+
+#ifdef TLS
+#define MODULE_NAME "encryption2"
+
+#include <resolv.h> /* base64 encode b64_ntop() and base64 decode b64_pton() */
+#include <sys/resource.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+/* Salt string length */
+#define PBKDF2_SALT_LEN 16 /* DO NOT TOUCH! */
+
+static Function *global = NULL;
+
+/* Cryptographic hash function used. openssl list -digest-algorithms */
+static char pbkdf2_method[28] = "SHA512";
+/* Enable re-encoding of password if pbkdf2-method and / or pbkdf2-rounds
+ * change.
+ */
+static int pbkdf2_re_encode = 0;
+/* Number of rounds (iterations). Default is 5000 like in glibc. */
+static int pbkdf2_rounds = 5000;
+
+static tcl_ints my_tcl_ints[] = {
+  {"pbkdf2-re-encode",  &pbkdf2_re_encode,  0},
+  {"pbkdf2-rounds",     &pbkdf2_rounds,     0},
+  {NULL,                NULL,               0}
+};
+
+static tcl_strings my_tcl_strings[] = {
+  {"pbkdf2-method", pbkdf2_method, 27, 0},
+  {NULL,            NULL,          0,  0}
+};
+
+static char *pbkdf2_close(void)
+{
+  return "You cannot unload the " MODULE_NAME " module.";
+}
+
+static void bufcount(char **buf, int *buflen, int bytes)
+{
+  *buf += bytes;
+  *buflen -= bytes;
+}
+
+static int b64_ntop_without_padding(u_char const *src, size_t srclength,
+                                    char *target, size_t targsize)
+{
+  char *c;
+
+  if (b64_ntop(src, srclength, target, targsize) < 0)
+    return -1;
+  c = strchr(target, '=');
+  if (c) {
+    *c = 0;
+    targsize = (c - target);
+  }
+  return targsize;
+}
+
+/* Return
```

Comment:
The name of the parameter "rounds" is "c" in the PBKDF2 specification, and https://www.npmjs.com/package/@phc/format uses "i". Where did "rounds" come from as parameter name?
I like that you found the PHC format, great find and great idea to use it

Reply:
Parapeter rounds comes from glibc https://github.com/lattera/glibc/blob/master/crypt/sha512-crypt.c#L83 see also man 3 crypt.